prowler/providers/azure/azure_provider.py

```diff
@@ -423,10 +453,22 @@ def test_connection(
 
         Raises:
             Exception: If failed to test the connection to Azure subscription.
+            AzureNoAuthenticationMethodError: If no authentication method is found.
+            AzureBrowserAuthNoTenantIDError: If browser authentication is enabled but the tenant ID is not found.
+            AzureTenantIDNoBrowserAuthError: If tenant ID is provided but browser authentication is not enabled.
+            AzureArgumentTypeValidationError: If there is an error in the argument type validation.
+            AzureSetUpRegionConfigError: If there is an error in setting up the region configuration.
+            AzureDefaultAzureCredentialError: If there is an error in retrieving the Azure credentials.
+            AzureInteractiveBrowserCredentialError: If there is an error in retrieving the Azure credentials using browser authentication.
+            AzureHTTPResponseError: If there is an HTTP response error.
+            SystemExit: If there is an error in the system.
```

Comment:
Are we still raising this?

Reply:
Of course, it could come from `validate_arguments`, `setup_region_config` and `setup_session`

Reply:
P.D: SystemExit deleted.